apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/ReadTests.cs

```diff
@@ -381,5 +376,56 @@ public ReadTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHelper
             Assert.Equal("b", (string) s1["C1"]);
             Assert.Equal(2, (long) s1["C2"]);
         }
+
+        [Fact]
+        public async Task TestDeadlineExceeded()
+        {
+            var oldTimeout = SpannerOptions.Instance.Timeout;
+            SpannerOptions.Instance.Timeout = TimeSpan.FromTicks(1);
+
+            try
+            {
+                //we use a new instance of a credential to force create a new spannerclient,
+                //which will cause the new options to be respected (ie timeout).
+                //normally setting timeout is only supported before creation of any client and cannot
+                //be changed due to the fact we pool clients.
+                var appDefaultCredentials = await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false);
+                if (appDefaultCredentials.IsCreateScopedRequired)
+                {
+                    appDefaultCredentials = appDefaultCredentials.CreateScoped(SpannerClient.DefaultScopes);
+                }
+
+                Exception exceptionCaught = null;
+                long result = 0;
+                try
```

Comment:
I'd use `var exception = await Assert.ThrowsAsync<SpannerException>(async() =>{ ... })` for this, rather than catching etc.